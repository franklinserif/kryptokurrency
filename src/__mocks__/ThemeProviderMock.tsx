import React, { ReactNode } from 'react';
import { mainTheme } from '../themes/mainTheme';
import { DefaultTheme, ThemeProvider } from 'styled-components';
import { ThemePropsType } from '../themes/Theme.types';
import { GlobalStyles } from '../themes/GlobalStyles';

/**
 * Theme component, it provide theme styles for all his childs components
 * @param {Object} props
 * @param {ReactNode} children
 * @returns {JSX.Element}
 */
const ThemeProviderMock: React.FC<ThemePropsType> = ({ children }: ThemePropsType) => {
  return (
    <ThemeProvider theme={mainTheme}>
      <GlobalStyles />
      {children}
    </ThemeProvider>
  );
};
export default ThemeProviderMock;
