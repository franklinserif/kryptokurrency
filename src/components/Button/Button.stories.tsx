/* eslint-disable react/jsx-props-no-spreading */

/* eslint-disable react/function-component-definition */
/* eslint-disable import/extensions */
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

// eslint-disable-next-line import/no-unresolved

import Button from './Button';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Button',
  component: Button,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof Button>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
// eslint-disable-next-line react/function-component-definition
const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const PrimaryButton = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
PrimaryButton.args = {
  variant: 'primary-button',
  children: 'Button',
};

export const SecondaryButton = Template.bind({});
SecondaryButton.args = {
  variant: 'secondary-button',
  children: 'Button',
};

export const TertiaryButton = Template.bind({});
TertiaryButton.args = {
  variant: 'tertiary-button',
  children: 'Button',
};

export const DangerButton = Template.bind({});
DangerButton.args = {
  variant: 'danger-button',
  children: 'Button',
};

export const GhostButton = Template.bind({});
GhostButton.args = {
  variant: 'ghost-button',
  children: 'Button',
};
