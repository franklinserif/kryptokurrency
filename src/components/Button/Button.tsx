import React from 'react';
import ButtonPropsTypes from './button.type';
import { StyledButton } from './button.styled';
/**
 * Button Components
 * @param {Object} props
 * @param {string} props.children
 * @param {string} props.variant
 * @param {handleClick: (event: React.MouseEvent<HTMLButtonElement>) => void} handleClick
 * @returns
 */
const Button = ({ children, variant, handleClick }: ButtonPropsTypes) => {
  return (
    <StyledButton variant={variant} onClick={handleClick}>
      {children}
    </StyledButton>
  );
};

export default Button;
