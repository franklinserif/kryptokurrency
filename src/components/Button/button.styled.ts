import styled from 'styled-components';
import { DefaultTheme } from 'styled-components';

const StyledButton = styled.button`
  background-color: ${(props: DefaultTheme) => props.colors.blue[100]};
`;
