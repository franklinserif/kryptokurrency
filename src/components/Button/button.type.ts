import React, { ReactNode } from 'react';
import { DefaultTheme } from 'styled-components';

type ButtonPropsTypes = {
  children: string;
  variant:
    | 'primary-button'
    | 'secondary-button'
    | 'tertiary-button'
    | 'danger-button'
    | 'ghost-button';
  handleClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
};

export default ButtonPropsTypes;
