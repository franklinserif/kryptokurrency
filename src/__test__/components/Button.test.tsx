/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow, mount, MountRendererProps } from 'enzyme';
import { create } from 'react-test-renderer';
import Button from '../../components/Button/Button';
import { DefaultTheme, ThemeProvider } from 'styled-components';
import ThemeProviderMock from '../../__mocks__/ThemeProviderMock';

describe('Button component', () => {
  const shallowWithTheme = (tree: React.ReactElement<any>, theme: DefaultTheme) => {
    const WrappingThemeProvider: React.FC<{ children: React.ReactChild }> = (props) => (
      <ThemeProvider theme={theme}>{props.children}</ThemeProvider>
    );

    return mount(
      tree,
      { wrappingComponent: WrappingThemeProvider } as MountRendererProps, //Override typing because @types/enzyme doesn't have wrappingComponent yet https://github.com/DefinitelyTyped/DefinitelyTyped/pull/36667
    );
  };
  /* 
  const button = shallowWithTheme(
    <Button variant="primary-button" handleClick={() => {}}>
      Text
    </Button>,
    mainTheme,
  ); */

  it('sould render correctly in debug mode', () => {
    const button = create(
      <ThemeProviderMock>
        <Button variant="primary-button" handleClick={() => {}}>
          Text
        </Button>
      </ThemeProviderMock>,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
});
