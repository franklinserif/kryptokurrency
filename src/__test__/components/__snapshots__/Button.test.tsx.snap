// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component sould render correctly in debug mode 1`] = `
.c0 {
  font-size: 15px;
  font-weight: 300;
  color: #fff;
  padding: 0.7rem 1.7rem;
  border: none;
  -webkit-transition: all 0.2s cubic-bezier(0.215,0.61,0.355,1);
  transition: all 0.2s cubic-bezier(0.215,0.61,0.355,1);
  background-color: #2b7dfa;
}

.c0:hover {
  cursor: pointer;
  box-shadow: 10px 10px 10px rgba(0,0,0,0.1);
}

.c0:active {
  box-shadow: inset 10px 10px 10px rgba(0,0,0,0.1);
}

<button
  className="c0"
  onClick={[Function]}
>
  Text
</button>
`;
